allauth/socialaccount/providers/agave/views.py

```diff
@@ -10,10 +11,9 @@
 
 class AgaveAdapter(OAuth2Adapter):
     provider_id = AgaveProvider.id
-    provider_default_url = 'https://public.agaveapi.co/'
-    provider_api_version = 'v2'
 
-    provider_base_url = 'https://public.agaveapi.co'
+    settings = app_settings.PROVIDERS.get(provider_id, {})
+    provider_base_url = settings.get("AGAVE_URL", 'https://public.agaveapi.co')
```

Comment:
How about naming this just `API_URL` -- the settings are already scoped with the provider ID `agave`, so this leads to needless `agave` names: `'agave': { 'AGAVE_URL': ...}`